### PR TITLE
Skip spotless tasks in Github actions to accelerate the CI

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Skip spotless tasks in Github actions to accelerate the CI

### Why

Optimization

### CODE REVIEW

File: .github/workflows/sub_gradle_test_and_build.yml [hunk 1]
```diff
@@ -47,7 +51,7 @@ jobs:
       - name: Gradle test and build. (unit tests, integration tests, end-2-end tests and build)
         env:
           run_docker: false
-        run: ./gradlew clean build --no-daemon --stacktrace
+        run: ./gradlew clean build --no-daemon --stacktrace -x spotlessApply -x spotlessCheck -x spotlessKotlin -x spotlessKotlinApply -x javadoc -x javadocJar -x sourcesJar
```
Comment (philipliu) [inline .github/workflows/sub_gradle_test_and_build.yml:54]: I think we can skip `spotlessApply`, but not check. If a developer tries to submit unformatted code, we should be catching it in CI.

Comment (lijamie98) [reply .github/workflows/sub_gradle_test_and_build.yml:54]: @philipliu If `spotlessCheck` fails, does it fail the CI?

Comment (philipliu) [reply .github/workflows/sub_gradle_test_and_build.yml]: Sorry, I missed this earlier, but we should keep `spotlessKotlin` too right?

Comment (lijamie98) [reply .github/workflows/sub_gradle_test_and_build.yml]: No problem at all. Thanks for catching these gotchas. I had removed `-x spotlessKotlin`.

**Pull request merged**